### PR TITLE
Rubicon: bidonmultiformat updates

### PR DESCRIPTION
**Pull request by And1sS**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java [hunk 1]
```diff
@@ -1760,10 +1789,11 @@ private Float cpmOverrideFromRequest(BidRequest bidRequest) {
     }
 
     private Float cpmOverrideFromImp(Imp imp) {
-        final ExtPrebid<ExtImpPrebid, ExtImpRubicon> extPrebid = imp != null ? parseRubiconExt(imp) : null;
-        final ExtImpRubicon bidder = extPrebid != null ? extPrebid.getBidder() : null;
-        final ExtImpRubiconDebug debug = bidder != null ? bidder.getDebug() : null;
-        return debug != null ? debug.getCpmoverride() : null;
+        return Optional.ofNullable(imp)
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:1792]: Imp can be null?

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java [hunk 2]
```diff
@@ -670,6 +693,12 @@ private RubiconImpExt makeImpExt(Imp imp,
                 .build();
     }
 
+    private Set<ImpMediaType> resolveFormats(ExtImpRubicon extImpRubicon, Set<ImpMediaType> impFormats) {
+        return Optional.ofNullable(extImpRubicon.getFormats())
+                .or(() -> Optional.ofNullable(extImpRubicon.getParams()).map(ExtImpRubiconParams::getFormats))
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:698]: `.map` on new line?

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java (hunk 2 above)
Comment (And1sS) [reply src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java]: Sorry for removed comment, missclick.

**Comment by muuki88** [comment]
Do I get this change right, that multi format bid requests will trigger a bid request to magnite for every media type?

**Comment by And1sS** [comment]
> Do I get this change right, that multi format bid requests will trigger a bid request to magnite for every media type?

Yes!

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**